apps/chat_server/lib/chat_server/contexts/rooms/update_room.ex

```diff
@@ -9,14 +10,15 @@ defmodule ChatServer.UpdateState do
     defstruct [:room]
   end
 
-  def call(params \\ %{}, _user, state) do
-    Logger.info "Updating room " <> inspect(params)
+  def call(room_id, _user, state) do
+    Logger.info "Updating room " <> inspect(room_id)
 
     # TODO verify user is allowed to update room of that type
 
-    with {:ok, record} <- update(params, state),
-         :ok <- broadcast_update(record) do
-      {:ok, record}
+    with {:ok, record} <- GetRoom(room_id),
```

Comment:
If you try to recompile the current code, you'll hit a syntax error that halts the execution. The culprit is this line. Instead of `GetRoom` it should be `GetRoom.call`.